app/oauth.go

```diff
@@ -574,19 +574,14 @@ func (a *App) LoginByOAuth(service string, userData io.Reader, teamId string) (*
 		return nil, model.NewAppError("LoginByOAuth", "api.user.login_by_oauth.parse.app_error",
 			map[string]interface{}{"Service": service}, "", http.StatusBadRequest)
 	}
-	authUser := provider.GetUserFromJson(bytes.NewReader(buf.Bytes()))
 
-	authData := ""
-	if authUser.AuthData != nil {
-		authData = *authUser.AuthData
-	}
-
-	if len(authData) == 0 {
+	authUser, err1 := provider.GetUserFromJson(bytes.NewReader(buf.Bytes()))
+	if err1 != nil || *authUser.AuthData == "" {
 		return nil, model.NewAppError("LoginByOAuth", "api.user.login_by_oauth.parse.app_error",
-			map[string]interface{}{"Service": service}, "", http.StatusBadRequest)
+			map[string]interface{}{"Service": service}, err1.Error(), http.StatusBadRequest)
```

Comment:
Is it possible for err1 to be nil and *authUser.AuthData == ""?

Reply:
It's a possibility. It escaped from me thanks @wiggin77  :)
```suggestion
			map[string]interface{}{"Service": service}, "", http.StatusBadRequest)
```